---
title: "Venetoclax resistance figures"
output:
  html_document:
    df_print: paged
---



```{r packages, echo=FALSE,message=FALSE}
library(synapser)
library(dplyr)
library(ggplot2)
library(bubbleHeatmap)
library(stringr)
library(umap)
synLogin()


##come up with a common color scheme for all cells
#https://coolors.co/palette/edae49-d1495b-00798c
colors=c(`Early resistant`='#EDAE49',Parental='#D1495B',`Late resistant`='#00798C')
dcolors=c(none='#CCCAAA',Gilteritinib='#EEDD55',`Gilteritinib/Venetoclax`='#0072B2')

```

This workbook will generate the figures for the papers


## Figure 1
This figure shows the data as it was collected 

```{r fig 1}

##genomics waterfall plot

##PCA of parental, g and GV samples, removed GVD and GD
global<-read.table(synGet('syn51197035')$path,fill=TRUE,header=T,row.names=NULL)|>
  subset(!is.na(value))|>
    subset(Treatment%in%(c('none','G','GV')))


phospho<-read.table(synGet('syn51197038')$path,fill=TRUE,header=T,row.names=NULL)|>
    subset(!is.na(value))|>
  subset(Treatment%in%(c('none','G','GV')))


gmat<-global|>dplyr::select(Sample,Gene,value)|>distinct()|>
  tidyr::pivot_wider(names_from=Sample,values_from=value,values_fill=0.0)|>tibble::column_to_rownames('Gene')
gmat<-gmat[-which(apply(gmat,1,function(x) any(x==0))),]

pmat<-phospho|>dplyr::select(Sample,SiteID,value)|>distinct()|>
  tidyr::pivot_wider(names_from=Sample,values_from=value,values_fill=0.0)|>tibble::column_to_rownames('SiteID')
pmat<-pmat[-which(apply(pmat,1,function(x) any(x==0))),]

meta<-global|>dplyr::select(Sample,Plex,State,Treatment,Ligand,Exp)|>distinct()|>
  mutate(Drug=ifelse(Treatment=='G',"Gilteritinib",ifelse(Treatment=='GV','Gilteritinib/Venetoclax','none')))|>
  mutate(State=ifelse(State=='Early',"Early resistant",ifelse(State=="Late","Late resistant",'Parental')))

meta$Exp[which(meta$Exp=='Gilteritinib,')]<-'Ex12'

gpcs<-prcomp(gmat)$rotation|>
    as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
  left_join(meta)

gu<-umap(t(gmat))$layout

colnames(gu)<-c('Dim1','Dim2')

gu|>  
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
right_join(meta)|>
  ggplot(aes(x=Dim1,y=Dim2,shape=State,col=Drug,size=5))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("UMAP of global proteomics")
  

gu|>  
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
right_join(meta)|>
  ggplot(aes(x=Dim1,y=Dim2,shape=Exp,col=Drug,size=5))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("UMAP of global proteomics")

ppcs<-prcomp(pmat)$rotation|>
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
  left_join(meta)

ggplot(gpcs,aes(x=PC1,y=PC2,shape=Exp,col=Drug,size=4))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("PCA of global proteomics")

ggplot(ppcs,aes(x=PC1,y=PC2,shape=Exp,col=Drug,size=4))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
    ggtitle("PCA of phospho proteomics")


pu<-umap(t(pmat))$layout

colnames(pu)<-c('Dim1','Dim2')

pu|>  
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
right_join(meta)|>
  ggplot(aes(x=Dim1,y=Dim2,shape=Exp,col=Drug,size=5))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("UMAP of phospho proteomics")

```

That weird experiment is an outlier
```{r exp 18}

meta<-subset(meta,Exp=='Ex18')
gpcs<-prcomp(gmat)$rotation|>
    as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
  right_join(meta)

gu<-umap(t(gmat))$layout

colnames(gu)<-c('Dim1','Dim2')


  

ggplot(gpcs,aes(x=PC1,y=PC2,shape=State,col=Drug,size=4))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("PCA of global proteomics")


gu|>  
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
right_join(meta)|>
  ggplot(aes(x=Dim1,y=Dim2,shape=State,col=Drug,size=5))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("UMAP of global proteomics")



ppcs<-prcomp(pmat)$rotation|>
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
right_join(meta)


ggplot(ppcs,aes(x=PC1,y=PC2,shape=State,col=Drug,size=4))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
    ggtitle("PCA of phospho proteomics")



pu<-umap(t(pmat))$layout

colnames(pu)<-c('Dim1','Dim2')

pu|>  
  as.data.frame()|>
  tibble::rownames_to_column('Sample')|>
right_join(meta)|>
  ggplot(aes(x=Dim1,y=Dim2,shape=State,col=Drug,size=5))+geom_point()+  scale_color_manual(values=dcolors)+theme_minimal()+
  ggtitle("UMAP of phospho proteomics")

```



## Figure 1 differential expression

```{r diffex analysis}
library(limma)

rownames(meta)<-NULL
meta<-meta|>tibble::column_to_rownames('Sample')
pars<-subset(meta,State=='Parental')
gvs<-subset(meta,Treatment=='GV')
gs<-subset(meta,Treatment=='G')


##first let's get gv diffex
gp<-pmat[,c(rownames(pars),rownames(gs))]
gg<-gmat[,c(rownames(pars),rownames(gs))]
design<- model.matrix(~ 0+factor(c(rep(1,nrow(pars)),rep(2,nrow(gs)))))
colnames(design)<-c('Parental','Gilteritinib')

contrasts.matrix<-cbind(Parental=c(1,0),Gilteritinib=c(0,1),'Gilt-Par'=c(-1,1))
                      
fit <- limma::lmFit(gp, design)
fit <- limma::contrasts.fit(fit, contrasts.matrix)
#browser()
fit <- limma::eBayes(fit)
pdeg <- limma::topTable(fit,
                       coef='Gilt-Par',
                                   adjust = "fdr",
                                   number = nrow(fit), 
                                sort.by = 'p')|>
  mutate(Drug='Gilteritinib',data='Phospho')


fit <- limma::lmFit(gg, design)
fit <- limma::contrasts.fit(fit, contrasts.matrix)
#browser()
fit <- limma::eBayes(fit)
gdeg <- limma::topTable(fit,coef='Gilt-Par',
                                   adjust = "fdr",
                                   number = nrow(fit), 
                                sort.by = 'p')|>
  mutate(Drug='Gilteritinib',data='Global')


alldeg<-rbind(pdeg,gdeg)|>
  tibble::rownames_to_column('Feature')

##then get gv diffex
gp<-pmat[,c(rownames(pars),rownames(gvs))]
gg<-gmat[,c(rownames(pars),rownames(gvs))]
design<- model.matrix(~ 0+factor(c(rep(1,nrow(pars)),rep(2,nrow(gvs)))))
colnames(design)<-c('Parental','GiltVen')

contrasts.matrix<-cbind(Parental=c(1,0),GiltVen=c(0,1),'GiltVen-Par'=c(-1,1))
                      
fit <- limma::lmFit(gp, design)
fit <- limma::contrasts.fit(fit, contrasts.matrix)
#browser()
fit <- limma::eBayes(fit)
pdeg <- limma::topTable(fit,
                       coef='GiltVen-Par',
                                   adjust = "fdr",
                                   number = nrow(fit), 
                                sort.by = 'p')|>
  mutate(Drug='Gilteritinib/Venetoclax',data='Phospho')


fit <- limma::lmFit(gg, design)
fit <- limma::contrasts.fit(fit, contrasts.matrix)
#browser()
fit <- limma::eBayes(fit)
gdeg <- limma::topTable(fit,coef='GiltVen-Par',
                                   adjust = "fdr",
                                   number = nrow(fit), 
                                sort.by = 'p')|>
  mutate(Drug='Gilteritinib/Venetoclax',data='Global')

alldeg<-rbind(pdeg,gdeg)|>
  tibble::rownames_to_column('Feature')|>
  rbind(alldeg)

stats<-alldeg|>subset(adj.P.Val<0.05)|>subset(abs(logFC)>1)|>group_by(data,Drug)|>summarize(nsig=n())
print(stats)
ggplot(stats,aes(x=Drug,y=nsig,fill=data))+geom_bar(stat='identity',position='dodge')
```

## Figure 2 enrichment

This figure shows the enrichment analysis.

```{r fig 2 plotting functions}

##load old library from github 
#devtools::install_github('PNNL-CompBio/amlresistancenetworks')
library(amlresistancenetworks)
gprots<-subset(alldeg,data=='Global')|>
  subset(Drug=='Gilteritinib')|>
  arrange(desc(logFC))|>
  dplyr::select(Gene='Feature',value='logFC')

gres<-computeGSEA(gprots,'gilteritinib',gsea_FDR=0.05)


gvprots<-subset(alldeg,data=='Global')|>
  subset(Drug=='Gilteritinib/Venetoclax')|>
  arrange(desc(logFC))|>
  dplyr::select(Gene='Feature',value='logFC')

gvres<-computeGSEA(gprots,'gilteritinib venetoclax',gsea_FDR=0.05)

####now do phospho
gphos<-subset(alldeg,data=='Phospho')|>
  subset(Drug=='Gilteritinib')|>
  arrange(desc(logFC))|>
  tidyr::separate(Feature,into=c('Gene','residue'),sep='-')|>
  dplyr::rename(value='logFC',p_adj='adj.P.Val')|>
  mutate(residue=gsub('[s|t|y]$','',residue))|>
  tidyr::separate_longer_delim(delim=stringr::regex('s|t|y'),cols='residue')
#  dplyr::select(Gene='Feature',value='logFC')

gkres<-computeKSEA(gphos,prefix='gilteritinib',ksea_FDR=0.05)


gvphos<-subset(alldeg,data=='Phospho')|>
  subset(Drug=='Gilteritinib/Venetoclax')|>
  arrange(desc(logFC))|>
  tidyr::separate(Feature,into=c('Gene','residue'),sep='-')|>
  dplyr::rename(value='logFC',p_adj='adj.P.Val')|>
  mutate(residue=gsub('[s|t|y]$','',residue))|>
  tidyr::separate_longer_delim(delim=stringr::regex('s|t|y'),cols='residue')
gvkres<-computeKSEA(gvphos,prefix='gilteritinib venetoclax',ksea_FDR=0.05)


```


```{r other analysis}

##from camilo
## Plotting function
plot_function <- function(enrichment_df, database, treatments, p_cutoff, n_pathways, str_width = 40){
  plot_title <- paste(treatments, collapse = ", ")
  enrichment_df <- enrichment_df %>%
    # filter(DB == database) %>%
    mutate(Pathway = gsub("_", " ", Pathway),
           Pathway = str_wrap(Pathway, width = str_width))
  
  enrichment_df_s <- enrichment_df %>%
    filter(p.adjust < p_cutoff,
           contrast %in% treatments) %>%
    mutate(sign = case_when(sign == "positive" ~ "o",
                            TRUE ~ "u")) %>%
    mutate(CS_ = paste(contrast, sign)) %>%
    mutate(CS_ = factor(CS_, levels = c(paste(treatments, "u"), paste(treatments, "o")))) %>%
    group_by(CS_) %>%
    top_n(n_pathways, -p.adjust) %>% ungroup()
  enrichment_df_s = rbind(enrichment_df_s %>% filter(sign == "o"), enrichment_df_s %>% filter(sign == "u"))
  pathways <- enrichment_df_s$Pathway %>% unique() 
  enrichment_df_s <- enrichment_df_s %>%
    mutate(Pathway = factor(Pathway, levels = pathways))

  if (length(unique(enrichment_df_s$Pathway)) >= 1){
    p <- ggplot(enrichment_df_s, aes(x = CS_, y = Pathway, size = Count, color = p.adjust)) + geom_point() + 
      scale_color_gradient(trans = 'log10', limits = c(min(enrichment_df_s$p.adjust),p_cutoff), low = "red", high = "blue") + 
      ggtitle(treatments) + theme(text = element_text(size = 15))
  } else {
    p <- ggplot() + ggtitle("Less than 1 pathways")
  }
  return(p)
}


new_plot_function <- function(enrichment_df, database, treatments, p_cutoff, n_pathways, str_width = 40){
  library(ggplotify)

  plot_title <- paste(treatments, collapse = ", ")
  enrichment_df <- enrichment_df %>%
    # filter(DB == database) %>%
    mutate(Pathway = gsub("_", " ", Pathway))#,
    #       Pathway = str_wrap(Pathway, width = str_width))
  
  enrichment_df_s <- enrichment_df %>%
    filter(p.adjust < p_cutoff,
           contrast %in% treatments) %>%
    mutate(sign = case_when(sign == "positive" ~ 1,
                            TRUE ~ -1)) %>%
    mutate(Count=Count*sign)|>
  #  mutate(CS_ = paste(contrast, sign)) %>%
  #  mutate(CS_ = factor(CS_, levels = c(paste(treatments, "u"), paste(treatments, "o")))) %>%
    group_by(contrast) %>%
    top_n(n_pathways/2, -p.adjust) %>% ungroup()
#  enrichment_df_s = rbind(enrichment_df_s %>% filter(sign == "o"), enrichment_df_s %>% filter(sign == "u"))
#  pathways <- enrichment_df_s$Pathway %>% unique() 
#  enrichment_df_s <- enrichment_df_s %>%
#    mutate(Pathway = factor(Pathway, levels = pathways))
  

  sizeMat=enrichment_df_s|>
    mutate(Significance=-log10(p.adjust))|>
    dplyr::select(Pathway,Significance,contrast)|>
    tidyr::pivot_wider(names_from=contrast,values_from=Significance,values_fill=0)|>
    tibble::column_to_rownames('Pathway')|>
    as.matrix()

  colorMat=enrichment_df_s|>
    dplyr::select(Pathway,Count,contrast)|>
    tidyr::pivot_wider(names_from=contrast,values_from=Count,values_fill=0)|>
    tibble::column_to_rownames('Pathway')|>
    as.matrix()
  
  res=bubbleHeatmap::bubbleHeatmap(t(colorMat),t(sizeMat),legendTitle=c('-log10 adj p-value','Count'))
  #grid.newpage()
  return(res)
  }


```


```{r fig2 plost}

ORA_global <- read.table(synGet('syn54958485')$path, sep = "\t", header = T)
ORA_RNA <- read.table(synGet("syn54958490")$path, sep = "\t", header = T)




ORA_df <- ORA_global %>% filter(DB == "GOBP") %>%
  mutate(contrast = sub("-none_Parental", "", contrast),
         Pathway = sub("GOBP ", "", ID))

p1 = new_plot_function(ORA_df, "gobp", c("GV_Early", "GV_Late"), 0.05, 40,50)


ORA_df <- ORA_global %>% filter(DB == "Hallmark") %>%
  mutate(contrast = sub("-none_Parental", "", contrast),
         Pathway = sub("HALLMARK_", "", ID),
         Pathway = gsub("_", " ", Pathway))
p2 = new_plot_function(ORA_df, "Hallmark", c("GV_Early", "GV_Late"), 0.2, 40,50)

ORA_df <- ORA_global %>% filter(DB == "KEGG") %>%
  mutate(contrast = sub("-none_Parental", "", contrast),
         Pathway = sub("KEGG ", "", ID))
p3 = new_plot_function(ORA_df, "KEGG", c("GV_Early", "GV_Late"), 0.05, 40,50)


ORA_df <- ORA_global %>% filter(DB == "Reactome") %>%
  mutate(contrast = sub("-none_Parental", "", contrast),
         Pathway = sub("REACTOME ", "", ID))
p4 = new_plot_function(ORA_df, "KEGG", c("GV_Early", "GV_Late"), 0.05, 40,50)


```

## Figure 3 

Cell type deconvolution

```{r fig 3}
tab<-readr::read_tsv(synGet('syn54488257')$path)

tab<-tab|>
  mutate(MOLM14=ifelse(is.na(State),'Parental',ifelse(State=='Early','Early resistant','Late resistant')))

p<-ggplot(tab,aes(x=`Cell type`,fill=MOLM14,y=value))+
  geom_boxplot()+
  scale_fill_manual(values=colors)+
  theme(axis.text.x = element_text(angle = 270, hjust = 0, vjust = 0.5))

p

ggsave('fig3_deconv.pdf',p,width=7,height=4)
ggsave('fig3_deconv.jpg',p,width=7,height=4)

```


## Figure 4

What    

