---
title: "Differential Expression Enrichment Analysis"
author: "Sara Gosline"
date: "2024-07-31"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(enrichR)
library(tidyr)
library(readr)
library(dplyr)
library(UpSetR)
library(ggfortify)
library(pheatmap)
```

## Load normalized data

We might as well redo the PCA plots here using ggplot and our annotated colors
```{r load matrix data}

prot_mat<-readr::read_csv('global_proteomics_matrix.csv')|>
  tibble::column_to_rownames('...1')|>
  as.matrix()

phos_mat<-readr::read_csv('phospho_proteomics_matrix.csv')|>
  tibble::column_to_rownames('...1')|>
  as.matrix()


##all metdata here
conds<-c("Decitabine Resistant","Early Gilteritinib Resistant","Late Gilteritinib Resistant","Early Gilteritinb/Decitabine Resistant","Late Gilteritinib/Decitabine Resistant",'Parental')
names(conds)=c('D_only','G_Early','G_Late','GD_Early','GD_Late','Parental')

colc<-c('#B75E5C','#E66100','#FFC20A','#0C7BDC','#44AA99','#AAAAAA')
names(colc)<-conds

dcolc<-colc[c(1,3,5,6)]
names(dcolc)=c('Decitabine','Gilteritinib','Gilteritinib/Decitabine','Parental')
samps<-colnames(prot_mat)

dvals<-lapply(samps,function(x) {
  if(length(grep('GD',x))>0)
    return('Gilteritinib/Decitabine')
  else if(length(grep('D_only',x))>0)
    return('Decitabine')
  else if(length(grep('G_',x))>0)
    return('Gilteritinib')
  else{
    return('Parental')
  }})|>
  unlist()

stage=lapply(samps,function(x) {
  if(length(grep('early',x,ignore.case = T))>0)
    return('Early')
  else if(length(grep('late',x,ignore.case=T))>0)
    return('Late')
  else{
    return('')
  }})|>
  unlist()

ddf<-data.frame(Sample=samps,drug=dvals,stage=stage)|>
  mutate(condition=stringr::str_trim(paste(stage,drug,ifelse(drug=='Parental','','Resistant'))))

propca=prcomp(t(prot_mat),scale=TRUE,center=TRUE)
phopca<-prcomp(t(phos_mat),scale=TRUE,center=TRUE)

autoplot(propca,data=ddf,color='drug',shape='stage',size=4)+scale_color_manual(values=dcolc)+theme_linedraw()
ggsave('proteinPCA.pdf')

autoplot(phopca,data=ddf,color='drug',shape='stage',size=4)+scale_color_manual(values=dcolc)+theme_linedraw()
ggsave('phosphoPCA.pdf')

```
## Load diffex 
First we load in the results of the diffex analysis performed in the previous step and summarize number of differentially expressed proteins and phosphosites.

First figure is just a barplot,  upsetplot of differentially expressed proteins and phosphosites across different conditions.

```{r load data,warning=F,error=F,echo=F}


##let's pick some standard colors for the conditions to use throughout the paper

phospho<-do.call(rbind,lapply(setdiff(names(conds),'Parental'),function(x){
  readr::read_csv(paste0('diff_ex_phospho_',x,'-none_Parental.csv'))|>
    mutate(condition=conds[x])
}))
phospho|>subset(adj.P.Val<0.05)|>group_by(condition)|>summarize(numFeatures=n())


prot<-do.call(rbind,lapply(setdiff(names(conds),'Parental'),function(x){
  readr::read_csv(paste0('diff_ex',x,'-none_Parental.csv'))|>
    mutate(condition=conds[x])
}))

prot|>subset(adj.P.Val<0.05)|>group_by(condition)|>summarize(numFeatures=n())

pdf('phosUpsetPlots.pdf',width=11)
phospho|>subset(adj.P.Val<0.05)|>dplyr::select(Var.1,condition)|>mutate(present=1)|>tidyr::pivot_wider(names_from=condition,values_from=present,values_fill=0)|>tibble::column_to_rownames('Var.1')|>upset(order.by='freq',sets.bar.color=colc[c(1,3,2,5,4)],text.scale=2)

dev.off()
#ggsave('phosUpsetPlots.pdf',p)
pdf('protUpsetPlot.pdf',width=11)
prot|>subset(adj.P.Val<0.05)|>dplyr::select(Var.1,condition)|>mutate(present=1)|>tidyr::pivot_wider(names_from=condition,values_from=present,values_fill=0)|>tibble::column_to_rownames('Var.1')|>upset(order.by='freq',sets.bar.color=colc[c(1,3,2,5,4)],text.scale=2)
dev.off()
```

## Protein enrichment

Now let's use enrichr to check out james' lists of interest

```{r enrichment of protein, echo=FALSE}

#here are the libaries that james used
elibs<-c('Reactome_2022',
         'KEGG_2021_Human',
    'GO_Biological_Process_2023',
 #   'HuBMAP_ASCTplusB_augmented_2022',
 #   'CellMarker_2024',
 #   'Azimuth_2023',
    'Human_Gene_Atlas',
#    'dbGaP',
#    'ChEA_2022',
#    'TRRUST_Transcription_Factors_2019',
#    'ENCODE_and_ChEA_Consensus_TFs_from_ChIP-X',
    'The_Kinase_Library_2023')

eof<-function(x){
  eres<-enrichr(x,elibs)
  do.call(rbind,lapply(names(eres),function(y){
    eres[[y]]|>
      mutate(pathway=y)
  }))
}
  
##evaluate which pathways are up/down regulated
eres<-prot|>
  subset(adj.P.Val<0.05)|>
  mutate(direction=ifelse(logFC>0,'up','down'))|>
  group_by(condition,direction)|>
  group_modify(~eof(.x$Var.1),.keep=TRUE)

readr::write_csv(eres,file='enrichmentResults.csv')

eres<-eres|>
  unite(c(condition,direction),col='test',sep=' ',remove=FALSE)

eres|>
  subset(Adjusted.P.value<0.05)|>
  group_by(pathway,condition,direction)|>summarize(numTerms=n_distinct(Term))|>
  ggplot(aes(x=pathway,y=numTerms,fill=condition))+geom_bar(stat='identity',position='dodge')+
  facet_grid(~direction)+
  scale_x_discrete(guide = guide_axis(angle=90))+
  scale_fill_manual(values=colc)

ggsave('numberOfTermsPerPathway.pdf')

for(path in unique(eres$pathway)){
tmap<-eres|>
    subset(pathway==path)|>
    subset(Adjusted.P.value<0.01)|>
    unite(c(condition,direction),col='test',sep=' ')|>
    dplyr::select(test,Term,Combined.Score)|>
    tidyr::pivot_wider(names_from=Term,values_from=Combined.Score,values_fill = 0.0)|>
    tibble::column_to_rownames('test')
    pheatmap(t(tmap),filename=paste0('termsFrom',path,'.pdf'))
}
 

##visualize them across the different conditons
```
Trying to plot individual terms is difficult, however the cell type information is informative.Let's talk about this in figure 2.

### Human gene atlas

let's look at those cell types
```{R cell types}
path='Human_Gene_Atlas'
tmap<-eres|>
    subset(pathway==path)|>
     ungroup()|>
    #subset(Adjusted.P.value<0.01)|>
    dplyr::select(test,Term,Combined.Score)|>
    tidyr::pivot_wider(names_from=Term,values_from=Combined.Score,values_fill = 0.0)|>
    tibble::column_to_rownames('test')
cols<-which(apply(tmap,2,function(x) length(which(x<10))<7))

metadata<-eres|>
  dplyr::select(condition,direction,test)|>
  distinct()|>
  tibble::column_to_rownames('test')
colors=list(condition=colc,direction=c(up='#EEEEEE',down='#000000'))

pheatmap(tmap[,cols],cellwidth = 10,cellheight=10,filename='cellTypeShift.pdf',annotation_row = metadata,annotation_colors=colors,show_rownames = F)

```

## Kinase enrichment

Now we can examine the phosphosites using KEA3 API


```{r phosphosite enrichment}
library(httr)
library(jsonlite)

doKEA<-function(genes){
  #genes = c("SMAD9","FOXO1","MYC","STAT1",'STAT3',"SMAD3")
  url = "https://maayanlab.cloud/kea3/api/enrich/"
  encode = "json"
  payload = list(query_name = "myQuery", gene_set = genes)

  response = POST(url = url, body = payload, encode = encode)
  json = content(response, "text")

  results = fromJSON(json)
  #print(results)
  integrated<-c('Integrated--meanRank','Integrated--topRank')
  others<-setdiff(names(results),integrated)
  
  rtab<-do.call(rbind,lapply(integrated,function(x)
                results[[x]]|>mutate(dset=x)))
  return(rtab)
}

kres<-phospho|>
  subset(adj.P.Val<0.1)|>
  mutate(direction=ifelse(logFC>0,'up','down'))|>
  group_by(condition,direction)|>
  group_modify(~doKEA(.x$Var.1),.keep=TRUE)

```