---
title: "patient outcomes cohort Data Summary"
author: "Sara Gosline"
date: "2025-06-25"
output: html_document
---

Here we can pull the metadata from Synapse to evaluate the numbers of samples and the overall data we have for those samples ahead of the patient outcomes study. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(synapser)
library(tidyverse)
library(readxl)
synapser::synLogin()
```

## Pull beatAML data

We have the initial beatAML data, including metabolomics and lipidomics, for beatAML samples to see what we have from there. 

```{r beataml data}
#all data
all_ba_data <- readxl::read_xlsx(synapser::synGet('syn68710501')$path) 

#patient data without proteomics sample numbers
ba_pats <- all_ba_data |>
  mutate(RNASeq=ifelse(is.na(dbgap_rnaseq_sample),FALSE,TRUE), Mutation = ifelse(is.na(dbgap_dnaseq_sample),FALSE,TRUE)) |>
  dplyr::select(labId,consensus_sex,reportedRace,reportedEthnicity,inferred_ethnicity,RNASeq, Mutation) |>
  distinct()


##now we have extra metadata that wasn't accounted for
ex_data <- readxl::read_xlsx(synGet('syn68731261')$path)|>
  mutate(RNASeq=FALSE,Mutation=FALSE) |>
dplyr::select(labId='Accession',consensus_sex='Sex',reportedRace,reportedEthnicity,
              inferred_ethnicity='inferredEthnicity',RNASeq, Mutation) |>
  distinct()

ba_pats <- rbind(ba_pats,ex_data)


#syn25796769 contains only metadata for the 210 samples with proteomics
all_ba_meta <- readxl::read_xlsx(synapser::synGet('syn25796769')$path)|>
  left_join(ba_pats,by = 'labId')|>
  dplyr::rename(CALR='Calreticulin',NPM1='NPM1calls')

##reformat BeatAML mutations to single gene name and categorical call. 
muts <- all_ba_meta[,c(3,6,7,75:207)]|>
  tidyr::pivot_longer(cols=c(-1),names_to='mutation',values_to='call')|> ##pull out all mutations into long table
  tidyr::separate(mutation,into=c('gene','rest'),sep='_') |> ##separate out method of calling mutation
  mutate(gene = ifelse(gene=='FLT3-ITDcalls','FLT3_ITD',gene)) |> ##rename FLT3 ITD
  mutate(newcall = ifelse(is.na(call),NA,ifelse(tolower(call) == 'negative',0,1))) |> ##now move to numeric
  group_by(labId,gene)|>
  summarize(mutation=max(newcall,na.rm = T))|>
  mutate(mutation=ifelse(is.finite(mutation),ifelse(mutation == 0,'WT','Mutant'),'Not measured'))


ba_meta <- all_ba_meta |>
  #mutate(Accession = labId) |>
  tidyr::separate(`SampleID(full)`, into = c("PTRC",'exp10','SampleID'),sep = '_') |>
  mutate(SampleID=as.numeric(SampleID))|>
  mutate(Race = ifelse(is.na(reportedRace),inferred_ethnicity,reportedRace))|> ##focus on reported RACE first
  mutate(Race = stringr::str_remove(Race,'Admixed')) |> ##remove the admixed 
  dplyr::select(Accession='labId', SampleID,'FLT3-ITDcalls',Sex='gender',Age='ageAtDiagnosis',Race,RNASeq, Mutation)|>
  mutate(Proteomics=TRUE,Phospho=TRUE)  ##all these patients have proteomics

 ba_mutation <- dplyr::select(ba_meta,c(labId='Accession',Sex,Race))|>left_join(muts)


#proteomics

#phospho

#lipidomics
ba_lipids <- readxl::read_xlsx(synapser::synGet('syn52121001')$path) |>
  tidyr::pivot_longer(cols = starts_with('BEAT_AML'), names_to = 'sample', values_to = 'value') |>
  tidyr::separate(sample, into = c('Beat','AML','PNL','SampleID.abbrev', 'l','posneg','date','other'), 
                  sep = '_') |>
  mutate(SampleID = as.numeric(SampleID.abbrev)) |>
  left_join(ba_meta, by = 'SampleID') |>
  dplyr::select('Average Mz','Metabolite name','Adduct type','Ontology','SampleID',
                'Accession')
#metabolomics
ba_metab <- readxl::read_xlsx(synapser::synGet('syn53678273')$path) |>
  tidyr::pivot_longer(cols = starts_with('BEAT_AML'), names_to = 'sample', values_to = 'value') |>
  tidyr::separate(sample, into = c('Beat','AML','PNL','SampleID.abbrev', 'm','hilic','posneg'), 
                  sep = '_') |>
  mutate(SampleID = as.numeric(SampleID.abbrev)) |>
  left_join(ba_meta, by = 'SampleID') |>
  dplyr::select('m/z','Name','Standardized name','Super class','Main class','SampleID',
                'Accession')


#rnaseq

##acetyl - what do we do with this data?
ba_ac <- read.table(synGet('syn53484040')$path,sep='\t',fill=TRUE,header=TRUE)
res<-apply(ba_ac,1,function(x) length(which(is.na(x))))

##filter for 50% present
less_missing <- ba_ac[which(res<110),]

mpats<- apply(less_missing,2,function(x) length(which(is.na(x))))

#filter patient for 50% ac sites
apats<-which(mpats<nrow(less_missing)/2)

##now get accession
acetyl_long<-less_missing[,apats] |>
  as.data.frame() |>
  tibble::rownames_to_column('acetylSite') |>
  tidyr::pivot_longer(cols=dplyr::starts_with("PTRC"),names_to='sId',values_to = 'value')|>
  tidyr::separate(sId,into=c('PTRC','exp','SampleID'),sep='_')|>
  mutate(SampleID=as.numeric(SampleID))|>
  left_join(ba_meta)


##let's collect all BA metadata
ba_meta <- ba_meta |>
  #mutate(Proteomics=labId%in%ba_meta$Accession, Phospho=labId%in%ba_meta$Accession) |>
  mutate(Metabolomics = Accession%in%ba_metab$Accession, Lipidomics = Accession%in%ba_lipids$Accession,Acetylomics = Accession%in%acetyl_long$Accession) |>
  mutate(source='BeatAML', study="BeatAML")
```

## Pull new metadata for pilot cohort

While we might not have the data yet, we have the manifests for additional samples we are collecting.

```{r new data}
library(ggplot2)

#lets get the colors straight
rcolors <- c(Black='orchid4',White='goldenrod',Unknown='ivory',
             HispNative='darkgreen',Asian='darkorange',Multiracial='darkred')


##new pilot metaadata
sample_exp_metadata <- readxl::read_xlsx(synGet('syn68522106')$path) |>
  dplyr::rename(CEBPA = 'CEBPA_BZIP',FLT3 = 'FLT3_TKD',Race='Race_label')|>
    mutate(Sex = ifelse(`Sex (1-male, 0-female)` == 1,'Male','Female'))|>
  tidyr::pivot_longer(cols=c(11:40),names_to='gene',values_to='mutation')|>
  mutate(mutation=ifelse(is.na(mutation),'Not measured',ifelse(mutation==0,'WT','Mutant')))

#mutaitons for genes
mutations <- unique(sample_exp_metadata$gene)#names(sample_exp_metadata)[11:40]
#mutations[which(mutations=='CEBPA_BZIP')]='CEBPA'

##filter so that we only have new samples, no beatAML
new_samp <- subset(sample_exp_metadata, !Accession %in% ba_meta$Accession)  |>
  dplyr::select(c('Accession','Age','Sex',Race,gene,mutation)) |>
  distinct()

#filter for betaaml data that is in the new cohort
ba <- subset(ba_meta,Accession %in% sample_exp_metadata$Accession)  |>
  dplyr::select(Accession, Age,Sex, Race,source) |>
  mutate(Race = ifelse(is.na(Race),'White',Race)) 

##how many samples by race and sex are there in each experiment? 
comb <- new_samp |>
  dplyr::select(Accession,Age,Sex,Race) |>
  mutate(source = 'newSamples')
##what is the age distribution of each experiment?

full_meta_pilot <- rbind(comb,ba) |>
  mutate(Study = 'pilotStudy')

full_meta_pilot |> ggplot(aes(x=source,fill=Race)) + facet_grid(~Sex) + geom_bar(position='dodge') + scale_y_log10() + scale_fill_manual(values=rcolors)
ggsave('pilotStudySamples.png',height=3)

plotlist <- lapply(c('Proteomics','Phospho',"Mutation","RNASeq",'Lipidomics','Acetylomics','Metabolomics'),function(x)
  ba_meta |> 
    subset(get(x)) |> ggplot(aes(x=source,fill=Race)) +facet_grid(~Sex) +  geom_bar(position='dodge') + scale_y_log10() + ggtitle(paste('Samples with',x,'data'))+scale_fill_manual(values=rcolors)
)

cowplot::plot_grid(plotlist=plotlist,nrow=2)
ggsave('beatAMLSamples.png',height=6,width=16)

##samples with data in both cohorts

plotlist <- lapply(c('Proteomics','Phospho',"Mutation","RNASeq",'Lipidomics','Metabolomics'),function(x)
  ba_meta |> subset(Accession %in% full_meta_pilot$Accession) |> 
    subset(get(x)) |> 
    data.frame()|>
    ggplot(aes(x=source,fill=Race)) +facet_grid(~Sex) +  geom_bar(position='dodge') + ggtitle(paste('Samples with',x,'data'))+scale_fill_manual(values=rcolors)
)

cowplot::plot_grid(plotlist=plotlist,nrow=2)
ggsave('beatAMLAndPilotSamples.png',height=6,width=16)


##now lets compare ages

pilot_age <- ggplot(full_meta_pilot,aes(x=Age,fill=Race))+geom_bar(position='dodge')+ggtitle("Pilot study age distribution")+scale_fill_manual(values=rcolors)
ba_age <- ggplot(ba_meta, aes(x=Age,fill=Race))+geom_bar(position='dodge')+ggtitle("BeatAMLstudy age distribution")+scale_fill_manual(values=rcolors)

cowplot::plot_grid(pilot_age,ba_age,nrow=2)

ggsave('combinedAgeDistribution.png',height=6)

```
## Mutation profiles
We also want to compare patients by mutational profile. Above we have pulled and harmonized mutation data for both cohorts, so now we just want to align and see what we can find for each cohort.

```{r mutations}

##now we can subset out just the mutation and race data and plot that...

#BA meta

ba_muts = ba_mutation|>
  dplyr::rename(Accession='labId')|>
  mutate(source='BeatAML',study='BeatAML')

#new meta
pilot_muts <- sample_exp_metadata|>
  dplyr::select(Accession,Sex,Race,gene,mutation)|>
  subset(!Accession%in%ba_muts$Accession)|>
  mutate(study='Pilot',source='Pilot')


full_muts <- rbind(ba_muts,pilot_muts)
full_muts

##we have a lot~

full_muts|>
  group_by(Race,gene,mutation)|>
  summarize(count=n())|>
  subset(gene%in%c("DNMT3A",'FLT3','FLT3_ITD',"IDH1","IDH2","KRAS","TP53","NPM1",'ATM')) |> ##what other genes do we want?
  ggplot(aes(x=mutation,y=count,fill=Race))+
  geom_bar(stat='identity',position='dodge')+facet_grid(~gene)+scale_fill_manual(values=rcolors)+theme(axis.text.x = element_text(angle = 90))

ggsave('mutationalProfiles.png',width=10,height=4)
```
## create new metadata

now we need to combine the data so that we can access the values in an easy fashion

```{r metadata}

full_meta <- ba_meta|>
  dplyr::select(Accession, Age, Sex, Race)|>
  mutate(source='BeatAML',Study='BeatAML')|>
  rbind(full_meta_pilot)
  
rmuts <- full_muts |>
  tidyr::pivot_wider(names_from=gene,values_from=mutation)|>
  dplyr::select(-c(Sex,Race,source,study))


meta_with_muts <- full_meta|>
  left_join(rmuts)

readr::write_csv(meta_with_muts,file='combinedBeatAML_pilotMetaData.csv')
f <- synapser::File('combinedBeatAML_pilotMetadata.csv',parentId='syn62750464')
synapser::synStore(f)
##store on synapse
```

## Pull/process metabolomics
We have the metabolomics meaurements in five tabs of this spreadsheet. this is for the pilot study analysis. 

```{r met}

mfile <- synGet('syn68710369')$path

hp <- readxl::read_xlsx(mfile,sheet = 'HILIC Positive') |>
  tidyr::pivot_longer(starts_with('PTRC'),names_to='Sample',values_to = 'value')|>
  mutate(Sample=stringr::str_replace(Sample,'QC_Pool_','QCPool'))|>  
  mutate(Sample=stringr::str_replace(Sample,'Blank_Pr_','BlankPr'))|>
  tidyr::separate(Sample,into=c('PTRC','exp','datatype','SampleID','runType','posneg'),sep='_')

hn <- readxl::read_xlsx(mfile,sheet = 'HILIC Negative')|>
  tidyr::pivot_longer(starts_with('PTRC'),names_to='Sample',values_to = 'value')|>
  mutate(Sample=stringr::str_replace(Sample,'QC_Pool_','QCPool'))|>  
  mutate(Sample=stringr::str_replace(Sample,'Blank_Pr_','BlankPr'))|>
  tidyr::separate(Sample,into=c('PTRC','exp','datatype','SampleID','runType','posneg'),sep='_')

rp <- readxl::read_xlsx(mfile,sheet = 'RP Positive')|>
  tidyr::pivot_longer(starts_with('PTRC'),names_to='Sample',values_to = 'value')|>
  mutate(Sample=stringr::str_replace(Sample,'QC_Pool_','QCPool'))|>  
  mutate(Sample=stringr::str_replace(Sample,'Blank_Pr_','BlankPr'))|>
  tidyr::separate(Sample,into=c('PTRC','exp','datatype','SampleID','runType','posneg'),sep='_')

rn <- readxl::read_xlsx(mfile,sheet = 'RP Negative')|>
  tidyr::pivot_longer(starts_with('PTRC'),names_to='Sample',values_to = 'value')|>
  mutate(Sample=stringr::str_replace(Sample,'QC_Pool_','QCPool'))|>  
  mutate(Sample=stringr::str_replace(Sample,'Blank_Pr_','BlankPr'))|>
  tidyr::separate(Sample,into=c('PTRC','exp','datatype','SampleID','runType','posneg'),sep='_')

full_metab <- rbind(hp,hn,rp,rn)


```
